routes/pages.js

```diff
@@ -0,0 +1,224 @@
+const express = require("express");
+const axios = require("axios");
+const swal = require("sweetalert");
+
+const router = new express.Router();
+
+const nyTimesNewsKeys = process.env.nyTimesNewsKeys;
+const ticketMasterKeys = process.env.ticketMasterKeys;
+const lastFMKey = process.env.lastFMKey;
+const documentaryKey = process.env.documentaryKey;
+
+
+
+
+router.get("/", async function (req, res, next) {
+  try {
+    return res.render("home.html");
+  } catch (err) {
+    return next(err);
+  }
+});
+
+
+router.get("/hip-hop", async function (req, res, next) {
+  try{
+     return res.render("./genres/hip-hop.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/r&b", async function (req, res, next) {
+  try{
+     return res.render("./genres/r&b.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/pop", async function (req, res, next) {
+  try{
+     return res.render("./genres/pop.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/blues", async function (req, res, next) {
+  try{
+     return res.render("./genres/blues.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/jazz", async function (req, res, next) {
+  try{
+     return res.render("./genres/jazz.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/rock", async function (req, res, next) {
+  try{
+     return res.render("./genres/rock.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+router.get("/country", async function (req, res, next) {
+  try{
+     return res.render("./genres/country.html");
+  }  catch (err) {
+     return next(err);
+  }
+});
+
+
+router.get("/artists", async function (req, res, next) {
+  try{
+      const artists_input = req.query.artists_input || "beyonce";
+      const url = `https://www.theaudiodb.com/api/v1/json/1/search.php?s=${artists_input.trim()}`;
+      const artists = await axios.get(url);
+      
+      const url2 = `http://ws.audioscrobbler.com/2.0/?method=artist.gettopalbums&artist=${artists_input.trim()}&limit=10&api_key=${lastFMKey}&format=json`;
+      const albums = await axios.get(url2);
+
+      const url3 = `http://ws.audioscrobbler.com/2.0/?method=artist.gettoptracks&artist=${artists_input.trim()}&limit=10&api_key=${lastFMKey}&format=json`;
+      const topSongs = await axios.get(url3);
+
+      const url4 = `http://ws.audioscrobbler.com/2.0/?method=artist.getsimilar&artist=${artists_input.trim()}&limit=5&api_key=${lastFMKey}&limit=4&format=json`;
+      const similarArtists = await axios.get(url4);
+
+      return res.render("./artists/artists.html", 
+      {img:artists.data.artists[0].strArtistThumb,
```

Comment:
what happens if some of these are undefined? Do you know they will always be defined? e.g. `artists.data.artists[0] = undefined`